template/src/app/api/get-token/route.ts

```diff
@@ -0,0 +1,88 @@
+import { NextResponse } from 'next/server';
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+type Text = {
+  access_token: string,
+  workspace_id: string,
+  error_description?: string
+};
+
+type Request = {
+  code: string
+};
+
+export async function POST(request: object) {
+  try {
+    const { code } = await request.json() as Request;
+
+    const {
+      CLIENT_ID: clientId,
+      CLIENT_SECRET: clientSecret,
+      REDIRECT_URI: redirectUri,
+      ORGANIZATION_ID: organizationId = '',
+      SOURCE_ID: sourceId = '',
+      SOURCE_BASE_URL: sourceBaseUrl
+    } = process.env;
+
+    const encoded = Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
+    const myHeaders = new Headers();
+    myHeaders.append('Authorization', `Basic ${encoded}`);
+    myHeaders.append('Content-Type', 'application/json');
+
+    const data = JSON.stringify({
+      'grant_type': 'authorization_code',
+      code,
+      'redirect_uri': redirectUri
+    });
+
+    const requestOptions = {
+      method: 'POST',
+      headers: myHeaders,
+      body: data
+    };
+
+    const res  = await fetch(`${sourceBaseUrl}/v1/oauth/token`, requestOptions);
```

Comment:
Declaring option directly inside fetch calls will ensure autocompletion, so it's prefferable to do this.
```suggestion
    const res  = await fetch(`${sourceBaseUrl}/v1/oauth/token`, {
      method: 'POST',
      headers: {
        Authorization: `Basic ${credentials}`,
        'Content-Type': 'application/json'
      },
      data: JSON.stringify({
        'grant_type': 'authorization_code',
        code,
        'redirect_uri': redirectUri
      })
    });
```

Reply:
Sure, Done